src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -1169,19 +1170,11 @@ static ucs_status_t uct_ib_mlx5dv_check_dc(uct_ib_device_t *dev)
         goto out_destroy_cq;
     }
 
-    qp_attr.send_cq              = cq;
-    qp_attr.recv_cq              = cq;
-    qp_attr.qp_type              = IBV_QPT_DRIVER;
-    qp_attr.comp_mask            = IBV_QP_INIT_ATTR_PD;
-    qp_attr.pd                   = pd;
-    qp_attr.srq                  = srq;
-
-    dv_attr.comp_mask            = MLX5DV_QP_INIT_ATTR_MASK_DC;
-    dv_attr.dc_init_attr.dc_type = MLX5DV_DCTYPE_DCT;
-    dv_attr.dc_init_attr.dct_access_key = UCT_IB_KEY;
+    uct_ib_mlx5dv_dct_qp_init_attr(&qp_init_attr, &dv_attr, pd, cq, srq);
 
     /* create DCT qp successful means DC is supported */
-    qp = UCS_PROFILE_CALL_ALWAYS(mlx5dv_create_qp, ctx, &qp_attr, &dv_attr);
+    qp = UCS_PROFILE_CALL_ALWAYS(mlx5dv_create_qp, ctx, &qp_init_attr,
+                                 &dv_attr);
```

Comment:
i'd remove the profile, as it is not a data-path

Reply:
it was added before my changes.
let's discuss and remove it in another PR